services/workflows-service/src/user/user.controller.internal.ts

```diff
@@ -67,5 +70,64 @@ export class UserControllerInternal {
       },
       projectIds?.[0] || currentProjectId,
     );
+
+    if (!options?.sendWelcomeEmail) {
+      return createdUser;
+    }
+
+    const message: sgMail.MailDataRequired = {
+      to: createdUser.email,
+      from: 'no-reply@ballerine.com',
+      subject: "Your Ballerine's Case Management System Credentials",
+      text: `
+      Dear ${userInfo.firstName} ${userInfo.lastName},
+
+        Welcome to the team! We are excited to have you on our system. To get you started, we have created your account for our Case Management System, which you will need to access to manage your tasks and collaborate with the team.
+      
+        To access your account:
+      
+        URL: ${env.BACKOFFICE_CORS_ORIGIN}
+        Email: ${userInfo.email}
+        Password: ${userInfo.password}
+      
+        If you encounter any issues accessing your account or have any questions, please do not hesitate to reach out to Ballerine's team.
+      
+        Best regards,
+        Ballerine's Team`,
+      html: `
+      <body>
+        <p>Dear ${userInfo.firstName} ${userInfo.lastName},</p>
+    
+        <p>Welcome to the team! We are excited to have you on our system. To get you started, we have created your account for our Case Management System, which you will need to access to manage your tasks and collaborate with the team.</p>
+    
+        <p><strong>To access your account:</strong></p>
+        <ul>
+          <li><strong>URL:</strong> ${env.BACKOFFICE_CORS_ORIGIN}</li>
+          <li><strong>Email:</strong> ${userInfo.email}</li>
+          <li><strong>Password:</strong> ${userInfo.password}</li>
+        </ul>
+    
+        <p>If you encounter any issues accessing your account or have any questions, please do not hesitate to reach out to Ballerine's team.</p>
+    
+        <p>Best regards,<br>
+        Ballerine's Team</p>
+      </body>`,
+    };
+
+    if (!env.SENDGRID_API_KEY) {
+      this.logger.warn('SendGrid API key not provided. Email will not be not send ');
+      this.logger.log('Email:', message);
+
+      return createdUser;
+    }
+
+    try {
+      sgMail.setApiKey(env.SENDGRID_API_KEY);
+      await sgMail.send(message);
+    } catch (error: Error | any) {
```

Comment:
If you union with `any` the type just becomes any. Error tends to be of type `unknown` since in JS even a string or a promise can be thrown, and when an error is thrown it may be of different types. Also, check out this instance I found in the codebase that seem to be closer to what you are trying to do here.

```
      this.logger.error(`Error sending email with SendGrid: ${isErrorWithMessage(error) ? error.message : ''}`, {
        error,
      });
```

And if you're familiar with type guards we have functions like `isErrorWithMessage`, or`isRecordNotFoundError`. These either validate the error has specific properties or that the error is of a specific instance.

```
export const PRISMA_RECORD_NOT_FOUND_ERROR = 'P2025';

export const isRecordNotFoundError = (
  error: unknown,
): error is Prisma.PrismaClientKnownRequestError & {
  code: typeof PRISMA_RECORD_NOT_FOUND_ERROR;
} =>
  error instanceof Prisma.PrismaClientKnownRequestError &&
  error.code === PRISMA_RECORD_NOT_FOUND_ERROR;
```

```
export interface IErrorWithMessage extends Error {
  message: string;
}

export const isErrorWithMessage = (error: unknown): error is IErrorWithMessage => {
  return isObject(error) && 'message' in error && typeof error.message === 'string';
};
```

Reply:
Thanks for pointing out, using a union type with `any` can diminish the benefits of type checking. I'll change the error type to `unknown` and commit.